<a href="https://colab.research.google.com/github/wshay3678/-datasci_1_loading/blob/main/HHA_507_Exercise_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import json

In [ ]:
df1 = '/content/Data /Inpatient_Rehabilitation_Facility-Conditions_Sep2026.csv'
read_df1 = pd.read_csv(df1)
read_df1.head(11)

,CMS Certification Number (CCN),Provider Name,Address Line 1,Address Line 2,City/Town,State,ZIP Code,County/Parish,Telephone Number,CMS Region,Condition,Count,Footnote
0,013025,ENCOMPASS HEALTH LAKESHORE REHABILITATION HOSP...,3800 RIDGEWAY DRIVE,-,BIRMINGHAM,AL,35209,Jefferson,(205) 868-2000,4,All other conditions,507,-
1,013025,ENCOMPASS HEALTH LAKESHORE REHABILITATION HOSP...,3800 RIDGEWAY DRIVE,-,BIRMINGHAM,AL,35209,Jefferson,(205) 868-2000,4,Brain disease or condition (non-traumatic),349,-
2,013025,ENCOMPASS HEALTH LAKESHORE REHABILITATION HOSP...,3800 RIDGEWAY DRIVE,-,BIRMINGHAM,AL,35209,Jefferson,(205) 868-2000,4,Brain injury (traumatic),67,-
3,013025,ENCOMPASS HEALTH LAKESHORE REHABILITATION HOSP...,3800 RIDGEWAY DRIVE,-,BIRMINGHAM,AL,35209,Jefferson,(205) 868-2000,4,Hip or femur fracture,245,-
4,013025,ENCOMPASS HEALTH LAKESHORE REHABILITATION HOSP...,3800 RIDGEWAY DRIVE,-,BIRMINGHAM,AL,35209,Jefferson,(205) 868-2000,4,"Hip or knee replacement, amputation or other b...",596,-
5,013025,ENCOMPASS HEALTH LAKESHORE REHABILITATION HOSP...,3800 RIDGEWAY DRIVE,-,BIRMINGHAM,AL,35209,Jefferson,(205) 868-2000,4,Nervous system disorder (excluding stroke),622,-
6,013025,ENCOMPASS HEALTH LAKESHORE REHABILITATION HOSP...,3800 RIDGEWAY DRIVE,-,BIRMINGHAM,AL,35209,Jefferson,(205) 868-2000,4,Spinal cord disease or condition (non-traumatic),85,-
7,013025,ENCOMPASS HEALTH LAKESHORE REHABILITATION HOSP...,3800 RIDGEWAY DRIVE,-,BIRMINGHAM,AL,35209,Jefferson,(205) 868-2000,4,Spinal cord injury (traumatic),26,-
8,013025,ENCOMPASS HEALTH LAKESHORE REHABILITATION HOSP...,3800 RIDGEWAY DRIVE,-,BIRMINGHAM,AL,35209,Jefferson,(205) 868-2000,4,Stroke,398,-
9,013028,ENCOMPASS REHABILITATION HOSPITAL OF MONTGOMERY,4465 NARROW LANE RD,-,MONTGOMERY,AL,36116,Montgomery,(334) 284-7821,4,All other conditions,449,-



| Feature | Description |
|---|---|
| **CMS Certification Number (CCN)** | A unique identification number assigned by CMS to the inpatient rehabilitation facility. |
| **Provider Name** | The official name of the inpatient rehabilitation facility. |
| **Address Line 1** | The primary street address of the facility. |
| **Address Line 2** | Additional address information, such as a suite or building number, when available. |
| **City/Town** | The city or town where the facility is located. |
| **State** | The two-letter abbreviation for the state where the facility is located. |
| **ZIP Code** | The ZIP code associated with the facility's address. |
| **County/Parish** | The county or parish in which the facility is located. |
| **Telephone Number** | The primary telephone number for the facility. |
| **CMS Region** | The CMS regional designation associated with the facility. |
| **Condition** | The rehabilitation condition or diagnostic category treated by the facility. |
| **Count** | The number of inpatient rehabilitation cases associated with the specified condition at the facility. |
| **Footnote** | Additional notes or explanatory information provided for the corresponding record. |

In [ ]:
df2 = '/content/Data /Inpatient_Rehabilitation_Facility-General_Information_Sep2026.csv'
read_df2 = pd.read_csv(df2)
read_df2.head(11)

,CMS Certification Number (CCN),Provider Name,Address Line 1,Address Line 2,City/Town,State,ZIP Code,County/Parish,Telephone Number,CMS Region,Ownership Type,Certification Date
0,013025,ENCOMPASS HEALTH LAKESHORE REHABILITATION HOSP...,3800 RIDGEWAY DRIVE,-,BIRMINGHAM,AL,35209,Jefferson,(205) 868-2000,4,For profit,10/01/1983
1,013028,ENCOMPASS REHABILITATION HOSPITAL OF MONTGOMERY,4465 NARROW LANE RD,-,MONTGOMERY,AL,36116,Montgomery,(334) 284-7821,4,For profit,05/11/1987
2,013029,ENCOMPASS HEALTH REHABILITATION HOSPITAL NORTH AL,1490 HIGHWAY 72 E,-,HUNTSVILLE,AL,35811,Madison,(205) 535-2300,4,For profit,06/04/1987
3,013030,ENCOMPASS HEALTH REHABILITATION HOSPITAL OF DO...,1736 EAST MAIN STREET,-,DOTHAN,AL,36301,Houston,(334) 712-6333,4,For profit,05/12/1994
4,013031,ENCOMPASS HEALTH REHABILITATION HOSPITAL OF SH...,900 OAK MOUNTAIN COMMONS LANE,-,PELHAM,AL,35124,Shelby,(205) 216-7621,4,For profit,05/25/2018
5,013032,ENCOMPASS HEALTH REHABILITATION HOSPITAL GADSDEN,801 GOODYEAR AVE,-,GADSDEN,AL,35903,Etowah,(256) 439-5000,4,For profit,08/14/2001
6,013033,REHABILITATION HOSPITAL OF PHENIX CITY,3715 US HIGHWAY 431 NORTH,-,PHENIX CITY,AL,36867,Russell,(334) 732-2200,4,For profit,11/12/2003
7,01T011,ST. VINCENTS EAST REHABILITATION CENTER,50 MEDICAL PARK EAST DRIVE,-,BIRMINGHAM,AL,35235,Jefferson,(205) 838-3672,4,Non-profit,04/30/1998
8,01T029,EAMC- LANIER,2000 PEPPERELL PARKWAY.,-,OPELIKA,AL,36801,Lee,(502) 596-6346,4,Government,10/01/2016
9,01T033,SPAIN REHABILITATION CENTER,SRC 223E,-,BIRMINGHAM,AL,35249,Jefferson,(205) 934-4940,4,Government,10/01/1997





| Feature | Description |
|---|---|
| **CMS Certification Number (CCN)** | A unique identification number assigned by the Centers for Medicare & Medicaid Services (CMS) to each inpatient rehabilitation facility. |
| **Provider Name** | The name of the inpatient rehabilitation facility or healthcare provider. |
| **Address Line 1** | The primary street address of the rehabilitation facility. |
| **Address Line 2** | Additional address information for the facility, such as a suite, floor, or building number. |
| **City/Town** | The city or town where the rehabilitation facility is located. |
| **State** | The state in which the rehabilitation facility is located, represented by a two-letter state abbreviation. |
| **ZIP Code** | The ZIP code associated with the rehabilitation facility's address. |
| **County/Parish** | The county or parish in which the rehabilitation facility is located. |
| **Telephone Number** | The primary telephone number for the rehabilitation facility. |
| **CMS Region** | The CMS region associated with the rehabilitation facility's location. |
| **Ownership Type** | Identifies the ownership classification of the rehabilitation facility, such as government, nonprofit, or proprietary ownership. |
| **Certification Date** | The date on which the facility received its CMS certification. |

In [ ]:
df3 = '/content/Inpatient_Rehabilitation_Facility-National_Data_Sep2026.csv'
read_df3 = pd.read_csv(df3)
read_df3.head(11)

,CMS Certification Number (CCN),Measure Code,Score,Footnote,Start Date,End Date,Measure Date Range
0,NATION,I_006_01_NATL_SIR,1.000,-,10/01/2024,09/30/2025,-
1,NATION,I_011_05_NATL_OBS_RATE,65.26,-,01/01/2025,12/31/2025,-
2,NATION,I_012_06_NATL_OBS_RATE,62.48,-,01/01/2025,12/31/2025,-
3,NATION,I_013_01_NATL_OBS_RATE,0.15,-,01/01/2025,12/31/2025,-
4,NATION,I_015_01_NATL_SIR,0.340,-,10/01/2024,09/30/2025,-
5,NATION,I_016_01_NATL_OBS_RATE,77.10,-,10/01/2024,03/31/2025,-
6,NATION,I_017_01_PPR_PD_NATL_OBS_RATE,9.43,-,10/01/2023,09/30/2025,-
7,NATION,I_017_01_PPR_PD_N_BETTER_NATL,31,-,10/01/2023,09/30/2025,-
8,NATION,I_017_01_PPR_PD_N_NO_DIFF_NATL,"1,084",-,10/01/2023,09/30/2025,-
9,NATION,I_017_01_PPR_PD_N_TOO_SMALL,14,-,10/01/2023,09/30/2025,-




| Feature | Description |
|---|---|
| **CMS Certification Number (CCN)** | The unique CMS identifier for the facility or provider. In this file, the value is `NATION`, indicating that the row represents national-level aggregate data rather than data for a specific facility. |
| **Measure Code** | The unique code identifying the quality or performance measure being reported (e.g., `I_006_01_NATL_SIR`, `I_017_01_PPR_PD_NATL_OBS_RATE`). The suffix often indicates the measure type (e.g., `SIR` = Standardized Infection Ratio, `OBS_RATE` = observed rate, `N_*` = counts by category). |
| **Score** | The numeric result or value for the measure. Depending on the measure, this may be a rate, ratio, percentage, or a count. Some values are quoted with commas (e.g., `"1,084"`). |
| **Footnote** | A symbol or code providing additional context or caveats about the score. In this file, all values are `-`, meaning no footnote applies. |
| **Start Date** | The beginning of the time period during which the data for the measure were collected (format: MM/DD/YYYY). |
| **End Date** | The end of the time period during which the data for the measure were collected (format: MM/DD/YYYY). |
| **Measure Date Range** | A field intended to describe the overall date range of the measure. In this file, all values are `-`, so the range must be inferred from the Start Date and End Date columns. |

In [ ]:
df4 = '/content/Data /supplier-diversity-regional-plan-2025-extract-1.xlsx'
read_df4 = pd.read_excel(df4)
read_df4.head(11)

,Hospital_Region_ID,Region_Name,Associated_Hospitals,System_Reporting_Organization,System_Description,System_Report_Start_Date,System_Report_End_Date,System_Require_Certification,System_Self_Certification,System_Tier_I_African_American,...,System_Tier_II_Disabled_Veteran,System_Total_Disabled_Veteran,System_Tier_I_Less_Duplicated_Amount,System_Tier_II_Less_Duplicated_Amount,System_Total_Less_Duplicated_Amount,System_Combined_Tier_I_Total,System_Combined_Tier_II_Total,System_Combined_Total,System_Total_Hospital_Procurement,System_Other_Relevant_Info
0,5,Palomar Health,"106370977,106374382",Palomar Health,"At Palomar Health, we believe that fostering a...",2025-01-01 08:00:00,2025-12-31 08:00:00,No,No,0.0,...,NaN,180272,2703171.0,NaN,2703171.0,3293049,NaN,3293049,56252495,NaN
1,67,Tenet Coachella Valley Market,"106331164,106331216,106362041",Tenet Coachella Valley Market,The Tenet Healthcare Coachella valley market c...,2025-01-01 08:00:00,2025-12-31 08:00:00,No,No,NaN,...,NaN,102500,2494137.0,NaN,2494137.0,10673100,NaN,10673100,111312821,NaN
2,18,Community Memorial Health System Total,"106560473,106560501",Community Memorial Health System,"Community Memorial Healthcare is a nonprofit, ...",2025-01-01 08:00:00,2025-12-31 08:00:00,Yes,No,0.0,...,0.0,0,48145.0,0.0,48145.0,419294,0.0,419294,178550078,NaN
3,3,Keck Medicine of USC,"106190529,106190818,106191216,106194219",Keck Medicine of USC,Keck Medicine of USC is the University of Sout...,2025-01-01 08:00:00,2025-12-31 08:00:00,Yes,Yes,2792195.0,...,60249.0,3923221,10233009.0,10748.0,10243757.0,44383700,630323.0,45014023,1426833514,NaN
4,1,Adventist Health System/West,"106100793,106100797,106150706,106150788,106154...",Adventist Health System/West,Includes spend for all Adventist Health System...,2025-01-01 08:00:00,2025-12-31 08:00:00,Yes,Yes,2883599.0,...,0.0,3065926,5394857.0,0.0,5394857.0,42261487,0.0,42261487,3348809291,Adventist Health System/West operates a shared...
5,68,Enloe Health,"106040828,106040962",Enloe Health,"Enloe Health is an independent, not-for-profit...",2025-01-01 08:00:00,2025-12-31 08:00:00,No,No,NaN,...,NaN,39817,NaN,NaN,NaN,525764,NaN,525764,185046172,"Small business spend $525,236\nSmall Disadvant..."
6,8,Sharp HealthCare System Report,"106370689,106370694,106370695,106370714,106370...",Sharp HealthCare,Sharp HealthCare System Report\nThis is a Syst...,2025-01-01 08:00:00,2025-12-31 08:00:00,Yes,No,367354.0,...,614.0,764,NaN,NaN,NaN,23259462,163269.0,23422731,2092332042,Sharp HealthCare's mission is to promote and p...
7,7,Scripps,"106370744,106370771,106371256,106371394",Scripps Health,"Scripps Memorial Encinitas, Scripps Memorial L...",2025-01-01 08:00:00,2025-12-31 08:00:00,Yes,No,1061824.0,...,81602.0,6353823,8793201.0,0.0,8793201.0,85521123,547858.0,86068981,831746795,NaN
8,70,NorthBay Medical Center and NorthBay VacaValle...,"106481357,106484001",NorthBay Medical Center and NorthBay VacaValle...,These two hospitals are located in Solano Coun...,2025-01-01 08:00:00,2025-12-31 08:00:00,Yes,No,660848.0,...,0.0,0,6017695.0,NaN,6017695.0,830756,0.0,830756,58297136,NorthBay Health maintains the same commitment ...
9,12,MARIAN REGIONAL MEDICAL CENTER - GROUP,"106400466,106420493",CommonSpirit Health,This system report includes data for the follo...,2025-01-01 08:00:00,2025-12-31 08:00:00,Yes,Yes,NaN,...,NaN,199690,NaN,NaN,NaN,408471,NaN,408471,204672837,Spend does not include items such as employee ...




| Feature | Description |
|---|---|
| **Hospital Region ID** | Unique identifier for the hospital region or health system. |
| **Region Name** | Name of the hospital region or health system. |
| **Associated Hospitals** | Hospitals included within the reported health system or region. |
| **Reporting Organization** | Organization or health system responsible for submitting the report. |
| **System Description** | General description of the hospital system or reporting organization. |
| **Reporting Period Start Date** | Beginning date of the reporting period. |
| **Reporting Period End Date** | Ending date of the reporting period. |
| **Supplier Certification Required** | Indicates whether suppliers are required to have formal certification. |
| **Supplier Self-Certification Accepted** | Indicates whether suppliers may certify their diversity status themselves. |
| **Tier I Spending with African-American Suppliers** | Procurement spending directly paid to African-American-owned businesses classified as Tier I suppliers. |
| **Tier II Spending with African-American Suppliers** | Procurement spending through Tier II suppliers involving African-American-owned businesses. |
| **Total Spending with African-American Suppliers** | Combined Tier I and Tier II spending with African-American-owned businesses. |
| **Tier I Spending with Hispanic-American Suppliers** | Procurement spending directly paid to Hispanic-American-owned businesses classified as Tier I suppliers. |
| **Tier II Spending with Hispanic-American Suppliers** | Procurement spending through Tier II suppliers involving Hispanic-American-owned businesses. |
| **Total Spending with Hispanic-American Suppliers** | Combined Tier I and Tier II spending with Hispanic-American-owned businesses. |
| **Tier I Spending with Native-American Suppliers** | Procurement spending directly paid to Native-American-owned businesses classified as Tier I suppliers. |
| **Tier II Spending with Native-American Suppliers** | Procurement spending through Tier II suppliers involving Native-American-owned businesses. |
| **Total Spending with Native-American Suppliers** | Combined Tier I and Tier II spending with Native-American-owned businesses. |
| **Tier I Spending with Asian-Pacific-American Suppliers** | Procurement spending directly paid to Asian-Pacific-American-owned businesses classified as Tier I suppliers. |
| **Tier II Spending with Asian-Pacific-American Suppliers** | Procurement spending through Tier II suppliers involving Asian-Pacific-American-owned businesses. |
| **Total Spending with Asian-Pacific-American Suppliers** | Combined Tier I and Tier II spending with Asian-Pacific-American-owned businesses. |
| **Tier I Spending with Unknown Minority Suppliers** | Tier I procurement spending with suppliers whose minority classification is recorded as unknown. |
| **Tier II Spending with Unknown Minority Suppliers** | Tier II procurement spending with suppliers whose minority classification is recorded as unknown. |
| **Total Spending with Unknown Minority Suppliers** | Combined Tier I and Tier II spending with suppliers classified as an unknown minority category. |
| **Total Tier I Minority Supplier Spending** | Total Tier I procurement spending across all reported minority-owned supplier categories. |
| **Total Tier II Minority Supplier Spending** | Total Tier II procurement spending across all reported minority-owned supplier categories. |
| **Total Minority Supplier Spending** | Combined procurement spending across Tier I and Tier II minority-owned suppliers. |
| **Tier I Spending with Women-Owned Suppliers** | Procurement spending directly paid to women-owned businesses classified as Tier I suppliers. |
| **Tier II Spending with Women-Owned Suppliers** | Procurement spending through Tier II suppliers involving women-owned businesses. |
| **Total Spending with Women-Owned Suppliers** | Combined Tier I and Tier II spending with women-owned businesses. |
| **Tier I Spending with LGBT-Owned Suppliers** | Procurement spending directly paid to LGBT-owned businesses classified as Tier I suppliers. |
| **Tier II Spending with LGBT-Owned Suppliers** | Procurement spending through Tier II suppliers involving LGBT-owned businesses. |
| **Total Spending with LGBT-Owned Suppliers** | Combined Tier I and Tier II spending with LGBT-owned businesses. |
| **Tier I Spending with Disabled-Veteran-Owned Suppliers** | Procurement spending directly paid to disabled-veteran-owned businesses classified as Tier I suppliers. |
| **Tier II Spending with Disabled-Veteran-Owned Suppliers** | Procurement spending through Tier II suppliers involving disabled-veteran-owned businesses. |
| **Total Spending with Disabled-Veteran-Owned Suppliers** | Combined Tier I and Tier II spending with disabled-veteran-owned businesses. |
| **Tier I Less-Duplicated Spending** | Tier I procurement spending calculated using the dataset's less-duplicated spending measure. |
| **Tier II Less-Duplicated Spending** | Tier II procurement spending calculated using the dataset's less-duplicated spending measure. |
| **Total Less-Duplicated Spending** | Total procurement spending calculated using the less-duplicated spending measure. |
| **Combined Tier I Spending** | Combined Tier I spending across the reported diverse supplier categories. |
| **Combined Tier II Spending** | Combined Tier II spending across the reported diverse supplier categories. |
| **Combined Diverse Supplier Spending** | Total procurement spending across Tier I and Tier II diverse suppliers. |
| **Total Hospital Procurement Spending** | Total procurement spending by the hospital system, including spending with diverse and other suppliers. |
| **Other Relevant Information** | Additional information supplied by the reporting hospital system. |

In [ ]:
df5 = '/content/Data /emergency-department-volume-and-capacity-2021-2023.xlsx'
read_df5 = pd.read_excel(df5)
read_df5.head(11)

,oshpd_id,FacilityName2,CountyName,system,year,LICENSED_BED_SIZE,HospitalOwnership,UrbanRuralDesi,TEACHINGDesignation,Category,Tot_ED_NmbVsts,EDStations,EDDXCount,LATITUDE,LONGITUDE,PrimaryCareShortageArea,MentalHealthShortageArea,Visits_Per_Station
0,106010735,Alameda Hospital,Alameda,Alameda Health System,2022,100-149,Government,Urban,Non-Teaching,Active COVID-19,13579,12.0,520,37.762660,-122.253991,No,No,43.333333
1,106010739,Alta Bates Summit Medical Center – Alta Bates ...,Alameda,Sutter Health,2022,300-499,Nonprofit,Urban,Non-Teaching,Active COVID-19,29760,22.0,1177,37.856450,-122.257430,No,No,53.500000
2,106010776,UC San Francisco Benioff Children's Hospital O...,Alameda,NaN,2022,150-199,Nonprofit,Urban,Teaching,Active COVID-19,35062,40.0,1822,37.837220,-122.267470,No,No,45.550000
3,106010846,Highland Hospital,Alameda,Alameda Health System,2022,200-299,Government,Urban,Non-Teaching,Active COVID-19,66095,53.0,2109,37.799250,-122.231380,No,No,39.792453
4,106010937,Alta Bates Summit Medical Center,Alameda,Sutter Health,2022,300-499,Nonprofit,Urban,Non-Teaching,Active COVID-19,29336,30.0,1325,37.821060,-122.262570,No,No,44.166667
5,106010987,Washington Hospital – Fremont,Alameda,NaN,2022,300-499,Government,Urban,Non-Teaching,Active COVID-19,43964,39.0,3013,37.558470,-121.980060,No,No,77.256410
6,106014050,Stanford Health Care Tri-Valley,Alameda,NaN,2022,150-199,Nonprofit,Urban,Non-Teaching,Active COVID-19,28620,17.0,1378,37.692060,-121.880950,No,No,81.058824
7,106014132,Kaiser Foundation Hospital – Fremont,Alameda,Kaiser Foundation Hospitals,2022,100-149,Nonprofit,Urban,Non-Teaching,Active COVID-19,38960,18.0,1487,37.550550,-121.974830,No,No,82.611111
8,106014233,Eden Medical Center,Alameda,Sutter Health,2022,100-149,Nonprofit,Urban,Non-Teaching,Active COVID-19,31062,22.0,1467,37.698377,-122.087406,No,No,66.681818
9,106014326,Kaiser Foundation Hospital – Oakland/Richmond,Alameda,Kaiser Foundation Hospitals,2022,300-499,Nonprofit,Urban,Non-Teaching,Active COVID-19,120088,80.0,4883,37.824020,-122.257910,No,No,61.037500



| Feature | Description |
|---|---|
| **Hospital ID** | Unique identifier assigned to the hospital or healthcare facility. |
| **Hospital Name** | Name of the hospital or healthcare facility. |
| **County** | County in which the hospital is located. |
| **Health System** | Health system or organization with which the hospital is affiliated. |
| **Reporting Year** | Year for which the hospital's emergency department data are reported. |
| **Licensed Bed Capacity** | Number of beds for which the hospital is licensed to provide care. |
| **Hospital Ownership Type** | Ownership classification of the hospital, such as nonprofit, government, or for-profit. |
| **Urban/Rural Classification** | Indicates whether the hospital is located in an urban or rural area. |
| **Teaching Hospital Designation** | Indicates the hospital's teaching status or designation. |
| **Hospital Category** | Classification or category assigned to the hospital in the dataset. |
| **Total Emergency Department Visits** | Total number of visits to the emergency department during the reporting year. |
| **Emergency Department Stations** | Number of treatment stations available in the emergency department. |
| **Emergency Department Diagnoses** | Number of diagnoses recorded during emergency department visits. |
| **Latitude** | Geographic latitude of the hospital's location. |
| **Longitude** | Geographic longitude of the hospital's location. |
| **Primary Care Shortage Area** | Indicates whether the hospital is located in or associated with an officially designated primary-care shortage area. |
| **Mental Health Shortage Area** | Indicates whether the hospital is located in or associated with an officially designated mental-health shortage area. |
| **Emergency Department Visits per Station** | Average number of emergency department visits per available treatment station, used to describe emergency department utilization relative to capacity. |

In [ ]:
df6 = '/content/Data /hci_unemployment_290_ca_re_co_cd_pl_ct-5-22-14-ada.xlsx'
read_df6 = pd.read_excel(df6)
read_df6.head(11)

,ind_id,ind_definition,reportyear,race_eth_code,race_eth_name,geotype,geotypevalue,geoname,county_fips,county_name,...,Unemployment,Labor_force,Unemployment_rate,ll_95ci,ul_95ci,se,rse,place_decile,ca_rr,version
0,290,Unemployment rate,2004,9.0,Total,CA,6.0,California,NaN,NaN,...,1089657.0,17444436.0,6.2,NaN,NaN,NaN,NaN,NaN,1.0,2014-05-22 11:47:47
1,290,Unemployment rate,2005,9.0,Total,CA,6.0,California,NaN,NaN,...,952559.0,17544763.0,5.4,NaN,NaN,NaN,NaN,NaN,1.0,2014-05-22 11:47:47
2,290,Unemployment rate,2006,9.0,Total,CA,6.0,California,NaN,NaN,...,865423.0,17686689.0,4.9,NaN,NaN,NaN,NaN,NaN,1.0,2014-05-22 11:47:47
3,290,Unemployment rate,2006-2010,1.0,AIAN,CA,6.0,California,NaN,NaN,...,9673.0,70286.0,13.8,12.608511,14.991489,0.607903,4.405092,NaN,1.0,2014-05-22 11:47:47
4,290,Unemployment rate,2006-2010,3.0,AfricanAm,CA,6.0,California,NaN,NaN,...,142776.0,1012992.0,14.1,13.742553,14.457447,0.182371,1.293410,NaN,1.0,2014-05-22 11:47:47
5,290,Unemployment rate,2006-2010,2.0,Asian,CA,6.0,California,NaN,NaN,...,175048.0,2452234.0,7.1,6.980851,7.219149,0.060790,0.856201,NaN,1.0,2014-05-22 11:47:47
6,290,Unemployment rate,2006-2010,4.0,Latino,CA,6.0,California,NaN,NaN,...,666317.0,6270852.0,10.6,10.480851,10.719149,0.060790,0.573493,NaN,1.0,2014-05-22 11:47:47
7,290,Unemployment rate,2006-2010,7.0,Multiple,CA,6.0,California,NaN,NaN,...,41077.0,320235.0,12.8,12.204255,13.395745,0.303951,2.374620,NaN,1.0,2014-05-22 11:47:47
8,290,Unemployment rate,2006-2010,5.0,NHOPI,CA,6.0,California,NaN,NaN,...,7636.0,69487.0,11.0,9.927660,12.072340,0.547112,4.973750,NaN,1.0,2014-05-22 11:47:47
9,290,Unemployment rate,2006-2010,8.0,Other,CA,6.0,California,NaN,NaN,...,5343.0,54725.0,9.8,8.608511,10.991489,0.607903,6.203089,NaN,1.0,2014-05-22 11:47:47




| Feature | Description |
|---|---|
| **Indicator ID** | Unique identifier for the unemployment indicator. |
| **Indicator Definition** | Description of what the unemployment indicator measures. |
| **Reporting Year** | Year in which the unemployment data were recorded or reported. |
| **Race/Ethnicity Code** | Code identifying the racial or ethnic population group represented in the observation. |
| **Race/Ethnicity** | Racial or ethnic group represented by the observation. |
| **Geographic Area Type** | Type of geographic area represented, such as state, county, or other local area. |
| **Geographic Area Type Code** | Code identifying the type of geographic area. |
| **Geographic Area Name** | Name of the geographic area represented by the observation. |
| **County FIPS Code** | Standard federal code used to identify the county associated with the observation. |
| **County Name** | Name of the county associated with the observation. |
| **Region Code** | Code identifying the geographic region. |
| **Region Name** | Name of the geographic region. |
| **Number of Unemployed People** | Estimated number of people who are unemployed within the specified population and geographic area. |
| **Labor Force** | Total number of people in the labor force, including both employed and unemployed individuals. |
| **Unemployment Rate** | Percentage of the labor force that is unemployed. |
| **Lower 95% Confidence Limit** | Lower boundary of the 95% confidence interval for the estimated value. |
| **Upper 95% Confidence Limit** | Upper boundary of the 95% confidence interval for the estimated value. |
| **Standard Error** | Measure of the statistical uncertainty associated with the estimate. |
| **Relative Standard Error** | Standard error expressed relative to the estimated value, indicating the reliability or precision of the estimate. |
| **Geographic Decile** | Decile ranking of the geographic area's unemployment measure compared with other geographic areas. |
| **California Relative Rank** | Measure comparing the geographic area's unemployment value with the corresponding California value. |
| **Data Version** | Date or version identifier indicating when the dataset was released or updated. |

In [ ]:
df7 = '/content/Data /MTA.json'
read_df7 = pd.read_json(df7)
read_df7.head(11)

,:id,:version,:created_at,:updated_at,fiscal_year,month,scenario,financial_plan_year,expense_type,agency,function,occupational_group,positions
0,row-d2a3_eedc_tsm6,rv-bkxv-7xuh~ca3i,2026-09-01 18:37:12.606000+00:00,2026-09-01 18:37:12.606000+00:00,2026,2026-07-01T00:00:00.000,Actual,2026,NREIMB,LIRR,Maintenance,Operational Hourlies,2738.30
1,row-6bhh.n5zt.k6je,rv-azkc-aswk.4fw2,2026-09-01 18:37:12.606000+00:00,2026-09-01 18:37:12.606000+00:00,2026,2026-07-01T00:00:00.000,Actual,2026,REIMB,MTABC,Engineering,Managers/Supervisors,11.00
2,row-zxfp-duyv.t6dx,rv-5tim.njhx~iueh,2026-09-01 18:37:12.606000+00:00,2026-09-01 18:37:12.606000+00:00,2026,2026-07-01T00:00:00.000,Actual,2026,REIMB,MNR,Public Safety,Managers/Supervisors,0.00
3,row-7zac~326i~ddvt,rv-tn97-gg6k~dghg,2026-09-01 18:37:12.606000+00:00,2026-09-01 18:37:12.606000+00:00,2026,2026-07-01T00:00:00.000,Actual,2026,REIMB,SIR,Maintenance,Managers/Supervisors,1.00
4,row-qfu8~pipn-63fh,rv-r95q~v5cu.pyy5,2026-09-01 18:37:12.606000+00:00,2026-09-01 18:37:12.606000+00:00,2026,2026-07-01T00:00:00.000,Actual,2026,NREIMB,SIR,Maintenance,Managers/Supervisors,31.00
5,row-j7mc_8855~q9ny,rv-ca9a_4xg3-kfdm,2026-09-01 18:37:12.606000+00:00,2026-09-01 18:37:12.606000+00:00,2026,2026-07-01T00:00:00.000,Actual,2026,REIMB,MNR,Administration,Managers/Supervisors,11.16
6,row-a66f_chud~i7za,rv-iqhx-er8e.z8a9,2026-09-01 18:37:12.606000+00:00,2026-09-01 18:37:12.606000+00:00,2026,2026-07-01T00:00:00.000,Actual,2026,REIMB,MTABC,Public Safety,Managers/Supervisors,0.00
7,row-6e5p-begb~di7r,rv-ywch-6xj3.7gkb,2026-09-01 18:37:12.606000+00:00,2026-09-01 18:37:12.606000+00:00,2026,2026-07-01T00:00:00.000,Actual,2026,REIMB,NYCT,Administration,Operational Hourlies,0.00
8,row-af8i_q744_ga29,rv-b6d5~ssyv-7mgs,2026-09-01 18:37:12.606000+00:00,2026-09-01 18:37:12.606000+00:00,2026,2026-07-01T00:00:00.000,Actual,2026,NREIMB,NYCT,Engineering,Managers/Supervisors,2.28
9,row-drku~kgyg_t7w8,rv-erni-pme4.qw77,2026-09-01 18:37:12.606000+00:00,2026-09-01 18:37:12.606000+00:00,2026,2026-07-01T00:00:00.000,Actual,2026,REIMB,MTABC,Public Safety,Operational Hourlies,0.00




| Feature | Description |
|---|---|
| **Record ID** | Unique identifier assigned to each record in the dataset. |
| **Record Version** | Version identifier associated with the individual record. |
| **Record Creation Date** | Date and time when the record was created. |
| **Record Last Updated Date** | Date and time when the record was most recently updated. |
| **Fiscal Year** | Fiscal year associated with the workforce data. |
| **Reporting Month** | Month for which the workforce position data are reported. |
| **Reporting Scenario** | Scenario under which the data are reported, such as actual results or a planning scenario. |
| **Financial Plan Year** | Financial plan year associated with the reported workforce data. |
| **Expense Type** | Classification of the expense associated with the reported positions, such as reimbursable or non-reimbursable expenses. |
| **MTA Agency** | MTA agency associated with the reported positions, such as NYCT, LIRR, Metro-North Railroad, or other MTA entities. |
| **Functional Area** | Primary organizational function in which the positions are categorized, such as Operations, Maintenance, Engineering, Administration, or Public Safety. |
| **Occupational Group** | Broad employee classification, such as Managers/Supervisors, Operational Hourlies, or Professional/Technical/Clerical employees. |
| **Number of Positions** | Number of workforce positions represented by the record for the specified agency, function, occupational group, and reporting period. |

In [ ]:
df8 = '/content/Data /airport.json'
read_df8 = pd.read_json(df8)
read_df8.head(11)

,:id,:version,:created_at,:updated_at,the_geom,objectid,locid,siteno,fullname,faa_st,...,stfips,stpostal,baselat,baselong,reclat,reclong,shape_stlength,rwylghts,srfctrtmnt,paveclas
0,row-jqus_bs8h-bqjg,rv-t9ct.kt8y-qebn,2025-07-22 03:31:27.956000+00:00,2025-07-22 03:31:27.956000+00:00,"{'type': 'MultiLineString', 'coordinates': [[[...",54,1N5,08645.01*A,BENNETT,MD,...,24,MD,38.412889,-75.623139,38.415611,-75.615889,894.948785,NaN,NaN,NaN
1,row-vd4u.6m6b.h4qd,rv-xy8m~bj5e.ek82,2025-07-22 03:31:27.956000+00:00,2025-07-22 03:31:27.956000+00:00,"{'type': 'MultiLineString', 'coordinates': [[[...",42,W00,08595.*A,FREEWAY,MD,...,24,MD,38.944644,-76.773103,38.938097,-76.771661,950.700290,MED,NaN,NaN
2,row-q3fd~gqkg~wpa6,rv-9jjr_7acm-ia58,2025-07-22 03:31:27.956000+00:00,2025-07-22 03:31:27.956000+00:00,"{'type': 'MultiLineString', 'coordinates': [[[...",44,OXB,08610.*A,OCEAN CITY MUNI,MD,...,24,MD,38.306456,-75.127728,38.315106,-75.125763,1246.539230,MED,NaN,NaN
3,row-mq7e~f4m2~mrjh,rv-dciv~w5id~vqhp,2025-07-22 03:31:27.956000+00:00,2025-07-22 03:31:27.956000+00:00,"{'type': 'MultiLineString', 'coordinates': [[[...",3,MTN,08452.*A,MARTIN STATE,MD,...,24,MD,39.332474,-76.422483,39.318849,-76.405047,2758.916899,HIGH,GRVD,NaN
4,row-sdyk~f4pm.dfqh,rv-29dt-iukq.zkgk,2025-07-22 03:31:27.956000+00:00,2025-07-22 03:31:27.956000+00:00,"{'type': 'MultiLineString', 'coordinates': [[[...",19,1W3,08512.*A,MEXICO FARMS,MD,...,24,MD,39.605717,-78.764169,39.606194,-78.756669,837.744672,LOW,NaN,NaN
5,row-7rax~a7q7.cd7d,rv-xkry_wf39~t76h,2025-07-22 03:31:27.956000+00:00,2025-07-22 03:31:27.956000+00:00,"{'type': 'MultiLineString', 'coordinates': [[[...",36,2W5,08577.*A,MARYLAND,MD,...,24,MD,38.599023,-77.073095,38.604078,-77.072167,727.420905,NaN,NaN,NaN
6,row-5rei-jn24~kz8g,rv-w7ts_4vwx_mpx7,2025-07-22 03:31:27.956000+00:00,2025-07-22 03:31:27.956000+00:00,"{'type': 'MultiLineString', 'coordinates': [[[...",46,NHK,08614.*A,PATUXENT RIVER NAS/TRAPNELL FIELD/,MD,...,24,MD,38.276562,-76.421032,38.290210,-76.418555,1955.012424,NaN,NaN,24 /R/B/W/T
7,row-prgy-gtwe-iimj,rv-7ya7-hrb6-hkpg,2025-07-22 03:31:27.956000+00:00,2025-07-22 03:31:27.956000+00:00,"{'type': 'MultiLineString', 'coordinates': [[[...",26,58M,08536.02*A,CECIL COUNTY,MD,...,24,MD,39.575948,-75.874566,39.572412,-75.865004,1180.701170,MED,AFSC,NaN
8,row-3msx_w86p_pqya,rv-zkyx~67iq-axg7,2025-07-22 03:31:27.956000+00:00,2025-07-22 03:31:27.956000+00:00,"{'type': 'MultiLineString', 'coordinates': [[[...",31,FDK,08556.*A,FREDERICK MUNI,MD,...,24,MD,39.423381,-77.379822,39.419055,-77.368366,1419.404249,MED,NaN,NaN
9,row-32zz_54n2_92tu,rv-pund~2w87.43gr,2025-07-22 03:31:27.956000+00:00,2025-07-22 03:31:27.956000+00:00,"{'type': 'MultiLineString', 'coordinates': [[[...",30,FDK,08556.*A,FREDERICK MUNI,MD,...,24,MD,39.409603,-77.380436,39.420499,-77.368440,2061.113491,HIGH,GRVD,NaN



| Feature | Description |
|---|---|
| **Record ID** | Unique identifier assigned to each dataset record. |
| **Record Version** | Version identifier associated with the individual record. |
| **Record Creation Date** | Date and time when the record was created. |
| **Record Last Updated Date** | Date and time when the record was most recently updated. |
| **Runway Geometry** | Geographic line geometry representing the physical location and orientation of the runway. |
| **Object ID** | Numeric identifier assigned to the runway feature. |
| **Airport Location ID** | Identifier for the airport or aviation facility associated with the runway. |
| **Airport Site Number** | Official site or facility number associated with the airport. |
| **Airport Name** | Full name of the airport or aviation facility. |
| **FAA State Code** | FAA abbreviation identifying the state where the airport is located. |
| **Runway Designation** | Runway number or designation indicating the runway's orientation, such as 04/22 or 15L/33R. |
| **Runway Length** | Length of the runway, reported in feet. |
| **Runway Width** | Width of the runway, reported in feet. |
| **Runway Surface Type** | Type and composition of the runway surface, such as asphalt, concrete, turf, or asphalt-concrete. |
| **Runway Surface Treatment** | Additional treatment or condition applied to the runway surface, such as grooving or other surface treatments. |
| **State FIPS Code** | Federal Information Processing Standards code identifying the state where the runway is located. |
| **State Postal Code** | Two-letter postal abbreviation for the state where the airport is located. |
| **Base End Latitude** | Latitude coordinate of the base or starting end of the runway. |
| **Base End Longitude** | Longitude coordinate of the base or starting end of the runway. |
| **Reciprocal End Latitude** | Latitude coordinate of the opposite or reciprocal end of the runway. |
| **Reciprocal End Longitude** | Longitude coordinate of the opposite or reciprocal end of the runway. |
| **Runway Geometry Length** | Calculated length of the runway geometry represented by the spatial feature. |

In [ ]:
df9 = '/content/Data /food-waste-pilot.json'
read_df9 = pd.read_json(df9)
read_df9.head(11)

,stop_name,date_collection_datetime,lbs_collected,compost_created_lbs
0,Citizen's Convenience Center,2026-04-29,515.8,52
1,Mills Park,2026-04-29,537.0,54
2,Mills Park,2026-04-27,561.0,56
3,Citizen's Convenience Center,2026-04-27,1093.0,109
4,Citizen's Convenience Center,2026-04-24,641.6,64
5,Mills Park,2026-04-24,573.0,57
6,Citizen's Convenience Center,2026-04-22,414.0,41
7,Mills Park,2026-04-22,811.4,81
8,Citizen's Convenience Center,2026-04-20,1259.8,126
9,Mills Park,2026-04-20,518.0,52




| Feature | Description |
|---|---|
| **Collection Site** | Name of the location where the food waste was collected, such as Citizen's Convenience Center or Mills Park. |
| **Collection Date** | Date when the food waste was collected. |
| **Food Waste Collected** | Total weight of food waste collected at the site, measured in pounds. |
| **Compost Created** | Amount of compost produced from the collected food waste, measured in pounds. |